example/poisson_gmg/parthenon_app_inputs.cpp

```diff
@@ -0,0 +1,110 @@
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#include <math.h>
+#include <sstream>
+#include <string>
+
+#include <parthenon/package.hpp>
+
+#include "config.hpp"
+#include "defs.hpp"
+#include "poisson_package.hpp"
+#include "utils/error_checking.hpp"
+
+using namespace parthenon::package::prelude;
+using namespace parthenon;
+
+// *************************************************//
+// redefine some weakly linked parthenon functions *//
+// *************************************************//
+
+namespace poisson_example {
+
+void ProblemGenerator(Mesh *pm, ParameterInput *pin, MeshData<Real> *md) {
+  auto pmb = md->GetBlockData(0)->GetBlockPointer();
+  const int ndim = md->GetMeshPointer()->ndim;
+
+  Real x0 = pin->GetOrAddReal("poisson", "x0", 0.0);
+  Real y0 = pin->GetOrAddReal("poisson", "y0", 0.0);
+  Real z0 = pin->GetOrAddReal("poisson", "z0", 0.0);
+  Real radius0 = pin->GetOrAddReal("poisson", "radius", 0.1);
+  Real interior_D = pin->GetOrAddReal("poisson", "interior_D", 1.0);
+  Real exterior_D = pin->GetOrAddReal("poisson", "exterior_D", 1.0);
+
+  auto desc =
+      parthenon::MakePackDescriptor<poisson_package::rhs, poisson_package::res_err,
+                                    poisson_package::u, poisson_package::D,
+                                    poisson_package::exact>(md);
+  auto pack = desc.GetPack(md);
+
+  constexpr auto te = poisson_package::te;
+  using TE = parthenon::TopologicalElement;
+  auto &cellbounds = pmb->cellbounds;
+  auto ib = cellbounds.GetBoundsI(IndexDomain::entire, te);
+  auto jb = cellbounds.GetBoundsJ(IndexDomain::entire, te);
+  auto kb = cellbounds.GetBoundsK(IndexDomain::entire, te);
+  pmb->par_for(
+      "Poisson::ProblemGenerator", 0, pack.GetNBlocks() - 1, kb.s, kb.e, jb.s, jb.e, ib.s,
+      ib.e, KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        const auto &coords = pack.GetCoordinates(b);
+        Real x1 = coords.X<1, te>(i);
+        Real x2 = coords.X<2, te>(j);
+        Real x3 = coords.X<2, te>(k);
+        Real x1f = coords.X<1, TE::F1>(i);
+        Real x2f = coords.X<2, TE::F2>(j);
+        Real x3f = coords.X<2, TE::F3>(k);
+        Real dx1 = coords.Dxc<1>(k, j, i);
+        Real dx2 = coords.Dxc<2>(k, j, i);
+        Real dx3 = coords.Dxc<3>(k, j, i);
```

Comment:
Again, from an "being an example" point of view, why the difference between single index versus triple index coordinates?
Not that it matters with the coordinates currently implemented, but if there's reason to believe that Dxc<3> changes with `j`, then x3f should probably also get a `j` dependence.

Reply:
that choice was totally arbitrary. It probably makes sense to always include all three indices to future proof against new coordinate systems.